threestudio/data/multiview.py

```diff
@@ -0,0 +1,379 @@
+        assert camera_dict["camera_model"] == "OPENCV"
+
+        frames = camera_dict["frames"]
+        frames = frames[::self.cfg.train_data_interval]
+        frames_proj = []
+        frames_c2w = []
+        frames_position = []
+        frames_direction = []
+        frames_img = []
+
+        self.frame_w = frames[0]["w"] // scale
+        self.frame_h = frames[0]["h"] // scale
+        print("Loading frames...")
+        self.n_frames = len(frames)
+
+        c2w_list = []
+        for frame in tqdm(frames):
+            extrinsic: Float[Tensor, "4 4"] = torch.as_tensor(frame["transform_matrix"], dtype=torch.float32)
+            c2w = extrinsic
+            c2w_list.append(c2w)
+        c2w_list = torch.stack(c2w_list, dim=0)
+
+        if self.cfg.camera_layout == "around":
+            c2w_list[:, :3, 3] -= torch.mean(c2w_list[:, :3, 3], dim=0).unsqueeze(0)
+        elif self.cfg.camera_layout == "front":
+            assert self.cfg.camera_distance > 0
+            c2w_list[:, :3, 3] -= torch.mean(c2w_list[:, :3, 3], dim=0).unsqueeze(0)
+            z_vector = torch.zeros(c2w_list.shape[0], 3, 1)
+            z_vector[:, 2, :] = -1
+            rot_z_vector = c2w_list[:, :3, :3] @ z_vector
+            rot_z_vector = torch.mean(rot_z_vector, dim=0).unsqueeze(0)
+            c2w_list[:, :3, 3] -= rot_z_vector[:, :, 0] * self.cfg.camera_distance
+        else:
+            raise ValueError(f"Unknown camera layout {self.cfg.camera_layout}. Now support only around and front.") 
+        
+
+        for idx, frame in tqdm(enumerate(frames)):
+            intrinsic: Float[Tensor, "4 4"] = torch.eye(4)
+            intrinsic[0, 0] = frame["fl_x"] / scale
+            intrinsic[1, 1] = frame["fl_y"] / scale
+            intrinsic[0, 2] = frame["cx"] / scale
+            intrinsic[1, 2] = frame["cy"] / scale
+            
+            frame_path = os.path.join(self.cfg.dataroot, frame["file_path"])
+            img = cv2.imread(frame_path)[:, :, ::-1].copy()
+            img = cv2.resize(img, (self.frame_w, self.frame_h))
+            img: Float[Tensor, "H W 3"] = torch.FloatTensor(img) / 255
+            frames_img.append(img)
+
+            direction: Float[Tensor, "H W 3"] = get_ray_directions(
+                self.frame_h, self.frame_w,
+                (intrinsic[0, 0], intrinsic[1, 1]),
+                (intrinsic[0, 2], intrinsic[1, 2]),
+                use_pixel_centers=False
+            )
+
+            c2w = c2w_list[idx]
+            camera_position: Float[Tensor, "3"] = c2w[:3, 3:].reshape(-1)
+
+            near = 0.1
+            far = 1000.0
+            proj = convert_proj(intrinsic, self.frame_h, self.frame_w, near, far)
+            proj: Float[Tensor, "4 4"] = torch.FloatTensor(proj)
+            frames_proj.append(proj)
+            frames_c2w.append(c2w)
+            frames_position.append(camera_position)
+            frames_direction.append(direction)
+        print("Loaded frames.")
```

Comment:
replace all `print` with `threestudio.info` or `threestudio.debug` to ensure the message only appears once in multi-node DDP. `threestudio.debug` will only show the message with `--verbose`